awscli/clidriver.py

```diff
@@ -434,68 +411,25 @@ def main(self, args=None):
+            args = self.prompt_for_args(args)
+            parsed_args, remaining = parser.parse_known_args(args)
             self._handle_top_level_args(parsed_args)
             self._emit_session_event(parsed_args)
             HISTORY_RECORDER.record(
                 'CLI_VERSION', self.session.user_agent(), 'CLI')
             HISTORY_RECORDER.record('CLI_ARGUMENTS', args, 'CLI')
             return command_table[parsed_args.command](remaining, parsed_args)
-        except PARAM_VALIDATION_ERRORS as e:
-            # RC 252 represents that the command failed to parse or failed
-            # client side validation at the botocore level.
-            LOG.debug("Client side parameter validation failed", exc_info=True)
-            write_exception(e, outfile=get_stderr_text_writer())
-            return PARAM_VALIDATION_ERROR_RC
-        except UnknownArgumentError as e:
-            sys.stderr.write("usage: %s\n" % USAGE)
-            sys.stderr.write(str(e))
-            sys.stderr.write("\n")
-            return PARAM_VALIDATION_ERROR_RC
-        except ConfigurationError as e:
-            # RC 253 represents that the command may be syntactically correct
-            # but the environment or configuration is incorrect.
-            LOG.debug("Invalid CLI or client configuration", exc_info=True)
-            write_exception(e, outfile=get_stderr_text_writer())
-            return CONFIGURATION_ERROR_RC
-        except NoRegionError as e:
-            msg = ('%s You can also configure your region by running '
-                   '"aws configure".' % e)
-            self._show_error(msg)
-            return CONFIGURATION_ERROR_RC
-        except NoCredentialsError as e:
-            msg = ('%s. You can configure credentials by running '
-                   '"aws configure".' % e)
-            self._show_error(msg)
-            return CONFIGURATION_ERROR_RC
-        except KeyboardInterrupt:
-            # Shell standard for signals that terminate
-            # the process is to return 128 + signum, in this case
-            # SIGINT=2, so we'll have an RC of 130.
-            sys.stdout.write("\n")
-            return 128 + signal.SIGINT
-        except ClientError as e:
-            # RC 254 represents that a request/response completed but the
-            # request failed for reasons specific to the service, returned
-            # by the service. Generally, this will indicate incorrect API
-            # usage and is likely not an issue with CLI.
-            LOG.debug("Service returned an exception", exc_info=True)
-            write_exception(e, outfile=get_stderr_text_writer())
-            return CLIENT_ERROR_RC
-        except Exception as e:
-            # RC 255 is the catch-all. 255 specifically should not be relied
-            # on as exceptions can move from this catch-all classification
-            # to a more specific RC such as one of the above.
-            LOG.debug("Exception caught in main()", exc_info=True)
```

Comment:
One thing that I missed from the previous review... We should definitely log the exception caught in `main()` and include the Traceback so that we can guarantee that we have a traceback for any exception in the logs. We can probably just take this line and paste it right before calling the `exception_handler.handle_exception` method